### DISCUSSION TITLE
Create a write only workload

### DISCUSSION THREAD

**Discussion by mdresch31** [TPC-Council/HammerDB#554] [Q&A] [answered]
Is there a way to make a write only workload with hammerDB? Or a majority write workload? 

I am testing against a PostgreSQL database using HammerDBCLI.

**Comment by sm-shaw** [comment, marked as answer]
To not ignore the obvious, the schema build itself is a write-only workload only inserting data into the tables. 

Otherwise, HammerDB is fully scripted so you can literally write any workload you want and run exactly the same script in the CLI or GUI. 

If you want to modify the current TPROC-C test you will need to save the driver script from the GUI first and load it into the CLI with the customscript command (Note a savescript command has already been added https://github.com/TPC-Council/HammerDB/pull/522 and will be in the next release or available now if building from source). Then to change the transaction mix either change the RandomNumber choice as shown below and/or comment-out the stored procedures you don't want to run, as shown below to not run the read-only transactions slev and ostat.

This way - you can change the transaction mix to modify the workload as you want and of course you can use PostgreSQL tools to edit the stored procedures themselves, so they run something different from standard, or write your own. 

To be clear just for illustration, in the modified script below we've changed choice to be a RandomNumber from 1 to 21 instead of the default 1 to 23 so 22 and 23 slev and ostat never get chosen.  Also we've shown commenting out slev and ostat which has the same effect if we left the RandomNumber at default in that 22 and 23 would be chosen but then not run anything when they do. 
```
 set choice [ RandomNumber 1 21 ]
            if {$choice <= 10} {
                if { $KEYANDTHINK } { keytime 18 }
                neword $lda $w_id $w_id_input $RAISEERROR $ora_compatible $pg_storedprocs
                if { $KEYANDTHINK } { thinktime 12 }
            } elseif {$choice <= 20} {
                if { $KEYANDTHINK } { keytime 3 }
                payment $lda $w_id $w_id_input $RAISEERROR $ora_compatible $pg_storedprocs
                if { $KEYANDTHINK } { thinktime 12 }
            } elseif {$choice <= 21} {
                if { $KEYANDTHINK } { keytime 2 }
                delivery $lda $w_id $RAISEERROR $ora_compatible $pg_storedprocs
                if { $KEYANDTHINK } { thinktime 10 }
            } elseif {$choice <= 22} {
                if { $KEYANDTHINK } { keytime 2 }
                #slev $lda $w_id $stock_level_d_id $RAISEERROR $ora_compatible $pg_storedprocs
                if { $KEYANDTHINK } { thinktime 5 }
            } elseif {$choice <= 23} {
                if { $KEYANDTHINK } { keytime 2 }
                #ostat $lda $w_id $RAISEERROR $ora_compatible $pg_storedprocs
                if { $KEYANDTHINK } { thinktime 5 }
            }
```
